### PR TITLE
No generated clientsets

### PR DESCRIPTION
**Pull request by EItanya**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/561

### CODE REVIEW

File: ci/check-code-and-docs-gen.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ make update-deps
 
 set +e
 
-REGENERATE_CLIENTS=1 make generated-code -B > /dev/null
```
Comment (harveyxia) [inline ci/check-code-and-docs-gen.sh:23]: Why are we removing this?

File: pkg/clients/zephyr/networking/access_control_policy.go [hunk 2]
```diff
@@ -3,31 +3,44 @@ package zephyr_networking
...
+	"k8s.io/client-go/rest"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type accessControlPolicyClient struct {
 	client client.Client
 }
 
+func NewAccessControlPolicyClientForConfig(cfg *rest.Config) (AccessControlPolicyClient, error) {
+	if err := networking_v1alpha1.AddToScheme(scheme.Scheme); err != nil {
```
Comment (harveyxia) [inline pkg/clients/zephyr/networking/access_control_policy.go:17]: Why was this introduced?

File: cli/pkg/tree/uninstall/crd/crd_remover.go [hunk 3]
```diff
@@ -23,36 +24,41 @@ var (
...
 
 	for _, crd := range crds.Items {
 		if strings.HasSuffix(crd.GetName(), cliconstants.ServiceMeshHubApiGroupSuffix) {
 			crdsDeleted = true
-			err := crdClient.Delete(crd.GetName())
-			if errors.IsNotFound(err) {
-				// may be a race condition elsewhere; the CRD was removed between when we listed it and when we came here to actually delete it
-				continue
-			} else if err != nil {
+			existing, err := crdClient.Get(ctx, crd.GetName())
```
Comment (GrahamGoudeau) [inline cli/pkg/tree/uninstall/crd/crd_remover.go:52]: why are we making these changes? I'd prefer to revert this section. We just did a list five lines previously, doing another get here isn't buying us anything

File: cli/pkg/tree/uninstall/uninstall_cmd.go [hunk 4]
```diff
@@ -184,7 +184,8 @@ func deregisterClusters(ctx context.Context, out io.Writer, kubeClusters *v1alph
 
 func removeManagementPlaneNamespace(ctx context.Context, out io.Writer, masterKubeClients *common.KubeClients, opts *options.Options) error {
 	if opts.SmhUninstall.RemoveNamespace {
-		err := masterKubeClients.NamespaceClient.Delete(ctx, opts.Root.WriteNamespace)
+
+		ns, err := masterKubeClients.NamespaceClient.Get(ctx, opts.Root.WriteNamespace)
```
Comment (GrahamGoudeau) [inline cli/pkg/tree/uninstall/uninstall_cmd.go:188]: again, not sure why we're adding this. It doesn't buy us anything that I can see. Prefer to revert

File: pkg/clients/kubernetes/core/interfaces.go [hunk 5]
```diff
@@ -26,7 +26,7 @@ type NodeClient interface {
 	List(ctx context.Context, options ...client.ListOption) (*corev1.NodeList, error)
 }
 
-type SecretsClient interface {
+type SecretClient interface {
```
Comment (GrahamGoudeau) [inline pkg/clients/kubernetes/core/interfaces.go:29]: 🙏

File: pkg/clients/zephyr/security/csr_client.go [hunk 6]
```diff
@@ -84,81 +94,3 @@ func (m *virtualMeshCSRClient) List(
...
-func (g *generatedVirtualMeshClient) UpdateStatus(
-	ctx context.Context,
-	vm *security_v1alpha1.VirtualMeshCertificateSigningRequest,
-	opts ...client.UpdateOption,
-) error {
-	updated, err := g.client.VirtualMeshCertificateSigningRequests(vm.Namespace).UpdateStatus(vm)
-	if err != nil {
-		return err
-	}
-	*vm = *updated
```
Comment (GrahamGoudeau) [inline pkg/clients/zephyr/security/csr_client.go:162]: can't comment on the file moves below, but: Revert the move of the errutils package

File: pkg/selector/resource_selector.go [hunk 7]
```diff
@@ -321,7 +318,7 @@ func getMeshServicesBySelectorNamespace(
 2) If namespaces is specified, the k8s must be in one of those namespaces
 3) The k8s Service must exist in the specified cluster. If cluster is empty, select across all clusters.
 */
-func kubeServiceMatches(
+func KubeServiceMatches(
```
Comment (GrahamGoudeau) [inline pkg/selector/resource_selector.go:321]: Why is this public now? Changing the visibility of a single previously private function from a component is not how I want us to be sharing code between here and the API server

File: cli/pkg/tree/uninstall/uninstall_cmd.go (hunk 4 above)
Comment (EItanya) [reply cli/pkg/tree/uninstall/uninstall_cmd.go:188]: If you look at the underlying implementation the dynamic client delete it takes the whole object.

File: cli/pkg/tree/uninstall/crd/crd_remover.go (hunk 3 above)
Comment (EItanya) [reply cli/pkg/tree/uninstall/crd/crd_remover.go:52]: The underlying impl of the dynamic client delete takes a runtime.Object, so we need the whole object. We can get it as part of the delete, or fetch it before

**Pull request merged**